### ISSUE TITLE
Flsimulate: Generate particles with specified energy spectrum (Falaise3)

### ISSUE THREAD

**Issue by noeroy** [SuperNEMO-DBD/Falaise#30]
Hello, could it be possible to add a feature that allows the user to generate particles, specifying the energy spectrum and the type of particle, for instance with a spectrum file? 
I've found this spectrum into old Falaise2 files that may have been used before by @cherylepatrick  if that could help illustrating what I ask: 

```
#@interpolation_name cspline ### used by the GSL library  
#@energy_unit MeV            ### for the first column  
1.6 3.37e-8
1.8 5.98e-7
2.0 2.95e-6
2.2 4.75e-6
2.4 6.56e-6
2.6 8.51e-6
2.8 1.05e-5
3.0 1.21e-5
3.2 1.33e-5
3.4 1.45e-5
3.6 1.61e-5
3.8 1.85e-5
4.0 2.29e-5
4.2 2.80e-5
4.4 3.43e-5
4.6 4.10e-5
4.8 4.80e-5
5.0 5.59e-5
5.2 6.48e-5
5.4 7.50e-5
5.6 8.62e-5
5.8 9.82e-5
6.0 1.10e-4
6.2 1.21e-4
6.4 1.33e-4
6.6 1.44e-4
6.8 1.56e-4
7.0 1.69e-4
7.2 1.83e-4
7.4 1.96e-4
7.6 2.10e-4
7.8 2.26e-4
8.0 2.42e-4
8.2 2.58e-4
8.4 2.74e-4
8.6 2.91e-4
8.8 3.10e-4
9.0 3.29e-4
9.2 3.49e-4
9.4 3.70e-4
9.6 3.91e-4
9.8 4.11e-4
```

**Comment by drbenmorgan** [comment]
I think this can be implemented as the "spectrum mode" functionality of `genbb::single_particle_generator`. That offers parameters:

- `data_file` : path to file containing spectrum data
- `mode` : `tabulated_function` or `histogram_pdf`, must be chosen to match format of input file
- `min_energy` : lower energy bound of sampling
- `max_energy` : upper bound of sampling

Do those supply enough configuration for the requirement?

**Comment by drbenmorgan** [comment]
Note that implementation of this will require review and merge of #29 first to confirm that's the correct way to make a new generator available to the user and to test it.

**Comment by noeroy** [comment]
Yes I think that would be enough.

**Comment by robobre** [comment]
I would like to ask if the feature that allows the user to generate particles, specifying the energy spectrum and the type of particle already exists. we need  to generate flat distribution in some energy range. it would be helpful in our simulations.

**Comment by drbenmorgan** [comment]
See #73 and PR #29

**Comment by cherylepatrick** [comment]
Hello! Can I check the status of this one please, @drbenmorgan @fmauger ? I think this will be what we need for the shielding analysis, but we might need additional help in working out how to use it.

**Comment by drbenmorgan** [comment]
See #75

**Comment by drbenmorgan** [comment]
Fixed by #75
